src/NuGet.Core/NuGet.Packaging/FnvHash64Function.cs

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace NuGet.Packaging
+{
+    /// <summary>
+    /// An Fnv 64-bit hash function that supports non-cryptographic hashing to optimize speed and minimize collisions.
+    /// 
+    /// This is non-private only to facilitate unit testing.
+    /// </summary>
+    public sealed class FnvHash64Function : IHashFunction
```

Comment:
This type is only used in the NuGet.ProjectModel assembly. It should be moved there. 

Given that it's specific to that one assembly, I think we can make the type internal as well. 
Some guidance about adding public APIs: https://github.com/NuGet/NuGet.Client/blob/dev/docs/nuget-sdk.md#changes-to-apis.

Reply:
I agree, let's move this to `NuGet.ProjectModel` and make it `internal`.

Reply:
I've address feedback for Genelle, please re-review